### ISSUE TITLE
`await supabase.auth.sign_up` raises `object NoneType can't be used in 'await' expression`

### ISSUE THREAD

**Issue by FilipeMarch** [supabase-community/supabase-py#654]
**Describe the bug**
`sign_up` method from `AsyncClient` is raising the error  `object NoneType can't be used in 'await' expression`

**To Reproduce**
```python
import os

import trio
from dotenv import load_dotenv

load_dotenv()


async def main():
    from supabase._async.client import AsyncClient, create_client

    DB_URL: str = os.getenv("DB_URL")
    ANON_KEY: str = os.getenv("ANON_KEY")

    supabase: AsyncClient = await create_client(DB_URL, ANON_KEY)  # works fine

    email: str = "test@gmail.com"
    password: str = "test123"

    user = await supabase.auth.sign_up({"email": email, "password": password})


trio.run(main)
```
```python
Traceback (most recent call last):
  File "/user/sbtest/test.py", line 23, in <module>
    trio.run(main)
  File "/user/sbtest/.venv/lib/python3.11/site-packages/trio/_core/_run.py", line 2093, in run
    raise runner.main_task_outcome.error
  File "/user/sbtest/quick.py", line 20, in main
    user = await supabase.auth.sign_up({"email": email, "password": password})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/sbtest/.venv/lib/python3.11/site-packages/gotrue/_async/gotrue_client.py", line 153, in sign_up
    await self._remove_session()
  File "/user/sbtest/.venv/lib/python3.11/site-packages/gotrue/_async/gotrue_client.py", line 653, in _remove_session
    await self._storage.remove_item(self._storage_key)
TypeError: object NoneType can't be used in 'await' expression
```

**Expected behavior**
Async sign up working

**Desktop (please complete the following information):**
 - Arch Linux
 - Python 3.11.6
 - Supabase 2.3.0

**Additional context**
Same issue on Android

**Comment by silentworks** [comment]
Thanks for opening this issue and also the PR you opened too. We have merged in a fix and it has been released in version 2.3.2.

**Comment by silentworks** [comment]
Closing this out as fix has been released.
